Build failure on Armbain GCC

Just grabbed latest and this what happens on Armbian on a NanoPi Fire3 building master.

[  1%] Built target ethash
[  4%] Built target argon2
[  5%] Building CXX object CMakeFiles/xmrig.dir/src/crypto/randomx/randomx.cpp.o
/root/xmrig/src/crypto/randomx/randomx.cpp: In member function ‘void RandomX_ConfigurationBase::Apply()’:
/root/xmrig/src/crypto/randomx/randomx.cpp:341:32: error: invalid operands of types ‘void (randomx::JitCompilerA64::*)(randomx::Instruction&, uint32_t&)’ {aka ‘void (randomx::JitCompilerA64::*)(randomx::Instruction&, unsigned int&)’} and ‘bool’ to binary ‘operator<’
   INST_HANDLE2(CBRANCH, CBRANCH<true>, FSQRT_R);
                                ^
/root/xmrig/src/crypto/randomx/randomx.cpp:285:96: note: in definition of macro ‘JIT_HANDLE’
 E(x, prev) randomx::JitCompilerA64::engine[k] = &randomx::JitCompilerA64::h_##x
                                                                               ^

/root/xmrig/src/crypto/randomx/randomx.cpp:341:3: note: in expansion of macro ‘INST_HANDLE2’
   INST_HANDLE2(CBRANCH, CBRANCH<true>, FSQRT_R);
   ^~~~~~~~~~~~
/root/xmrig/src/crypto/randomx/randomx.cpp:300:57: error: expected primary-expression before ‘;’ token
  for (; k < freq_sum; ++k) { JIT_HANDLE(func_name, prev); }
                                                         ^
/root/xmrig/src/crypto/randomx/randomx.cpp:341:3: note: in expansion of macro ‘INST_HANDLE2’
   INST_HANDLE2(CBRANCH, CBRANCH<true>, FSQRT_R);
   ^~~~~~~~~~~~
/root/xmrig/src/crypto/randomx/randomx.cpp:344:32: error: invalid operands of types ‘void (randomx::JitCompilerA64::*)(randomx::Instruction&, uint32_t&)’ {aka ‘void (randomx::JitCompilerA64::*)(randomx::Instruction&, unsigned int&)’} and ‘bool’ to binary ‘operator<’
   INST_HANDLE2(CBRANCH, CBRANCH<false>, FSQRT_R);
                                ^
/root/xmrig/src/crypto/randomx/randomx.cpp:285:96: note: in definition of macro ‘JIT_HANDLE’
 E(x, prev) randomx::JitCompilerA64::engine[k] = &randomx::JitCompilerA64::h_##x
                                                                               ^

/root/xmrig/src/crypto/randomx/randomx.cpp:344:3: note: in expansion of macro ‘INST_HANDLE2’
   INST_HANDLE2(CBRANCH, CBRANCH<false>, FSQRT_R);
   ^~~~~~~~~~~~
/root/xmrig/src/crypto/randomx/randomx.cpp:300:57: error: expected primary-expression before ‘;’ token
  for (; k < freq_sum; ++k) { JIT_HANDLE(func_name, prev); }
                                                         ^
/root/xmrig/src/crypto/randomx/randomx.cpp:344:3: note: in expansion of macro ‘INST_HANDLE2’
   INST_HANDLE2(CBRANCH, CBRANCH<false>, FSQRT_R);
   ^~~~~~~~~~~~
make[2]: *** [CMakeFiles/xmrig.dir/build.make:2351: CMakeFiles/xmrig.dir/src/crypto/randomx/randomx.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:74: CMakeFiles/xmrig.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
root@nanopifire3:~/xmrig/build#

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,6 +24,7 @@ option(WITH_ADL             "Enable ADL (AMD Display Library) or sysfs support (
 option(WITH_STRICT_CACHE    "Enable strict checks for OpenCL cache" ON)
 option(WITH_INTERLEAVE_DEBUG_LOG "Enable debug log for threads interleave" OFF)
 option(WITH_PROFILING       "Enable profiling for developers" OFF)
+option(WITH_SSE             "Enable SSE for Blake2" ON)
 
 option(BUILD_STATIC         "Build static binary" OFF)
 option(ARM_TARGET           "Force use specific ARM target 8 or 7" 0)

--- a/cmake/randomx.cmake
+++ b/cmake/randomx.cmake
@@ -65,7 +65,7 @@ if (WITH_RANDOMX)
         set_property(SOURCE src/crypto/randomx/jit_compiler_a64_static.S PROPERTY LANGUAGE C)
     endif()
 
-    if (CMAKE_C_COMPILER_ID MATCHES GNU OR CMAKE_C_COMPILER_ID MATCHES Clang)
+    if (NOT ARM_TARGET AND WITH_SSE AND (CMAKE_C_COMPILER_ID MATCHES GNU OR CMAKE_C_COMPILER_ID MATCHES Clang))
         set_source_files_properties(src/crypto/randomx/blake2/blake2b_sse41.c PROPERTIES COMPILE_FLAGS -msse4.1)
     endif()
 

--- a/doc/build/CMAKE_OPTIONS.md
+++ b/doc/build/CMAKE_OPTIONS.md
@@ -22,6 +22,7 @@ This feature add external dependency to libhwloc (1.10.0+) (except MSVC builds).
 * **`-DWITH_EMBEDDED_CONFIG=ON`** Enable [embedded](https://github.com/xmrig/xmrig/issues/957) config support.
 * **`-DWITH_OPENCL=OFF`** Disable OpenCL backend.
 * **`-DWITH_CUDA=OFF`** Disable CUDA backend.
+* **`-DWITH_SSE=OFF`** Disable SSE for Blake2 (useful for arm builds).
 
 ## Debug options
 

